wfe/wfe.go

```diff
@@ -1840,8 +1840,12 @@ func (wfe *WebFrontEndImpl) orderForDisplay(
 	defer order.RUnlock()
 
 	// Copy the initial OrderRequest from the internal order object to mutate and
-	// use as the result.
+	// use as the result. We have to make sure to copy the identifiers and
+	// authorizations *slices* and not *pointers to the slices* or the mutations
+	// below could mutate the order in the database, causing data races.
 	result := order.Order
+	result.Identifiers = slices.Clone(order.Order.Identifiers)
+	result.Authorizations = slices.Clone(order.Order.Authorizations)
```

Comment:
I must admit that I'm not certain if this is the "right", Goish, Gothonic way to fix this, so I'm happy to be shown better solutions.

Reply:
This definitely works. But it has the downside of not catching any new pointer-valued fields which are added to the Order type in the future.

I think it would be slightly more idiomatic to instead say:
```go
result := acme.Order{
	Status: order.Order.Status,
	Error: order.Order.Error,
  etc
	Identifiers: slices.Clone(order.Order.Identifiers),
  etc
}
```
This has a similar downside -- if a new field is added to the Order type, it may not get added to this operation. But that will result in the field being _missing_, rather than subtler bugs, so is much more likely to be caught by unit or integration tests.

This solution is so much more verbose that I'm honestly not sure it counts as "better", but I do think it has worthwhile advantages. Up to you!

Reply:
Yeah, the thing where unmentioned struct fields get filled with the zero value kinda works against us here. I like yours better because then at least we're not mutating an object after creating it, but I don't know how to do better.